go/enclave/components/batch_executor.go

```diff
@@ -110,6 +110,7 @@ func (executor *batchExecutor) ComputeBatch(context *BatchExecutionContext) (*Co
 
 	// we need to copy the batch to reset the internal hash cache
 	copyBatch := *batch
+	copyBatch.ResetHash()
```

Comment:
I'd move this to below the mutation for clarity (line 115). That should be the pattern if we're using this Reset method, that every time properties are set the ResetHash gets called afterwards.